integration-tests/smoke/ccip/ccip_test.go

```diff
@@ -206,3 +208,797 @@ func TestTokenTransfer(t *testing.T) {
+		linkAddress := state.Chains[dest].LinkToken.Address()
+		feeQuoter := state.Chains[dest].FeeQuoter
+		timestampedPrice, err := feeQuoter.GetTokenPrice(nil, linkAddress)
+		require.NoError(t, err)
+		require.Equal(t, changeset.MockLinkPrice, timestampedPrice.Value)
+
+		// Wait for all exec reports to land
+		changeset.ConfirmExecWithSeqNrsForAll(t, e, state, expectedSeqNumExec, startBlocks)
+
+		balance, err := dstToken.BalanceOf(nil, state.Chains[tenv.FeedChainSel].Receiver.Address())
+		require.NoError(t, err)
+		require.Equal(t, twoCoins, balance)
+
+		// Delete from the expcted seq num the chain that was just tested so that we don't pass it to the
+		// commit report in the next test
+		delete(expectedSeqNum, sourceDestPair)
+	})
+
+	t.Run("Send message Pay with native remote chain -> home", func(t *testing.T) {
+		// Assign Src to the Home Chain Selector and destChain to the Feed Chain Selector
+		src := tenv.FeedChainSel
+		dest, destChain := tenv.HomeChainSel, e.Chains[tenv.HomeChainSel]
+
+		sourceDestPair := changeset.SourceDestPair{
+			SourceChainSelector: src,
+			DestChainSelector:   dest,
+		}
+
+		// get the header for the destination chain and the relevant block number
+		latesthdr, err := destChain.Client.HeaderByNumber(testcontext.Get(t), nil)
+		require.NoError(t, err)
+		block := latesthdr.Number.Uint64()
+		startBlocks[dest] = &block
+
+		// Send to the receiver on the destination chain paying with native token
+		var (
+			receiver = common.LeftPadBytes(state.Chains[dest].Receiver.Address().Bytes(), 32)
+			data     = []byte("hello world")
+			feeToken = common.HexToAddress("0x0")
+		)
+
+		ccipMessage := router.ClientEVM2AnyMessage{
+			Receiver:     receiver,
+			Data:         data,
+			TokenAmounts: emptyTokenArray,
+			FeeToken:     feeToken,
+			ExtraArgs:    nil,
+		}
+
+		msgSentEvent := changeset.TestSendRequest(t, e, state, src, dest, false, ccipMessage)
+		expectedSeqNum[changeset.SourceDestPair{
+			SourceChainSelector: src,
+			DestChainSelector:   dest,
+		}] = msgSentEvent.SequenceNumber
+		expectedSeqNumExec[changeset.SourceDestPair{
+			SourceChainSelector: src,
+			DestChainSelector:   dest,
+		}] = []uint64{msgSentEvent.SequenceNumber}
+
+		// Wait for all commit reports to land.
+		changeset.ConfirmCommitForAllWithExpectedSeqNums(t, e, state, expectedSeqNum, startBlocks)
+
+		// After commit is reported on all chains, token prices should be updated in FeeQuoter.
+		linkAddress := state.Chains[dest].LinkToken.Address()
+		feeQuoter := state.Chains[dest].FeeQuoter
+		timestampedPrice, err := feeQuoter.GetTokenPrice(nil, linkAddress)
+		require.NoError(t, err)
+		require.Equal(t, changeset.MockLinkPrice, timestampedPrice.Value)
+
+		// Wait for all exec reports to land
+		changeset.ConfirmExecWithSeqNrsForAll(t, e, state, expectedSeqNumExec, startBlocks)
+
+		delete(expectedSeqNum, sourceDestPair)
+	})
+
+	t.Run("Send message pay with wrapped native home chain -> remote", func(t *testing.T) {
```

Comment:
We can do something like this:
1. Setup one lane say home chain -> remote
2. Create test input struct with fee token type (link, native and wnative) and message type (Token, Data and Token+Data)
3. Loop the test and try to run in parallel using multiple wallet address. 

WDYT?